jobs/haproxy/templates/drain.erb

```diff
@@ -6,13 +6,21 @@ set -e
 pidfile=/var/vcap/sys/run/bpm/haproxy/haproxy.pid
 sockfile=/var/vcap/sys/run/haproxy/stats.sock
 logfile=/var/vcap/sys/log/haproxy/drain.log
+lockfile=/var/vcap/sys/run/haproxy/drain.lock
+
+mkdir -p "$(dirname ${logfile})"
```

Comment:
this was actually unrelated to the lock file but I noticed that in the template, in case drain is disabled, we actually write to the log file before the `mkdir` call, so it could potentially fail. So, to correct that I've moved the `mkdir` before the first log statement. (I actually think the `mkdir` may be obsolete because BOSH should create the `/var/vcap/sys/log/haproxy/` directory regardless, however I didn't verify that).